client/extensions/woocommerce/state/ui/orders/edits/test/reducer.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+import reducer from '../reducer';
+import {
+	WOOCOMMERCE_UI_ORDERS_CLEAR_EDIT,
+	WOOCOMMERCE_UI_ORDERS_EDIT,
+} from 'woocommerce/state/action-types';
+
+describe( 'reducer', () => {
+	it( 'should store the currently editing order', () => {
+		const action = {
+			type: WOOCOMMERCE_UI_ORDERS_EDIT,
+			siteId: 123,
+			order: {
+				id: 40,
+				billing: {
+					first_name: 'Joan',
+				}
+			},
+		};
+		const newState = reducer( undefined, action );
+		expect( newState ).to.eql( {
+			currentlyEditingId: 40,
+			changes: {
+				billing: {
+					first_name: 'Joan',
+				}
+			},
+		} );
+	} );
+
+	it( 'should update the order when it\'s changed', () => {
+		const action = {
+			type: WOOCOMMERCE_UI_ORDERS_EDIT,
+			siteId: 123,
+			order: {
+				id: 40,
+				billing: {
+					first_name: 'Joan',
+					last_name: 'Watson',
+				}
+			},
+		};
+		const originalState = deepFreeze( {
+			currentlyEditingId: 40,
+			changes: {
+				billing: {
+					first_name: 'Joan',
+				}
+			},
+		} );
+		const newState = reducer( originalState, action );
+		expect( newState ).to.eql( {
+			currentlyEditingId: 40,
+			changes: {
+				billing: {
+					first_name: 'Joan',
+					last_name: 'Watson',
+				}
+			},
+		} );
+	} );
+
+	it( 'should merge updates to an order if new fields are passed in', () => {
```

Comment:
Probably need to add a similar test for not-yet-persisted-to-the-server orders. Please consider comment above about whether placeholder ID should be rolling or not (that might be a good thing to test itself.)

Reply:
Oh I see now - the placeholder ID is passed back to the reducer via the action, so there should be no difference between newly created or edited orders, but I've added the test anyway.